code/lspgen/src/lspgen_packet.c

```diff
@@ -450,7 +450,23 @@ lspgen_serialize_isis_attr(lsdb_attr_t *attr, lsdb_packet_t *packet)
         case ISIS_TLV_EXTD_IS_REACH:
             push_data(buf, attr->key.link.remote_node_id, 7);
             push_be_uint(buf, 3, attr->key.link.metric); /* Metric */
-            push_be_uint(buf, 1, 0); /* subTLV length */
+	    if(attr->key.link.adjacency_sid == true) {
+                push_be_uint(buf, 1, 7); /* subTLVs length */
+		push_be_uint(buf, 1, ISIS_SUBTLV_IS_EXT_ADJ_SID);
+		push_be_uint(buf, 1, 5);
+		push_be_uint(buf, 1, 0x30); /* set V and L flag always */
+		push_be_uint(buf, 1, 0);
+                /* ensure proper random seed */
+                struct timespec ts;
+                clock_gettime(CLOCK_MONOTONIC, &ts);
+                srand((time_t)ts.tv_nsec);
+		push_be_uint(buf, 1, 0);
+                /* generate random adjacency SID but avoid reserved ranges 0-255 */
+	        push_be_uint(buf, 2, (uint16_t) rand() % 3840 + 256);
```

Comment:
Even when using the optimal seed value, there remains a potential risk of two adjacencies sharing the same SID.

Reply:
Please see latest commit, I changed to set a base value for adjacency-sid instead of random generation. Works for both lspgen and bbl.